pkg/albingress/albingress.go

```diff
@@ -151,6 +155,18 @@ func NewALBIngressFromIngress(o *NewALBIngressFromIngressOptions) *ALBIngress {
 		Tags:                 newIngress.Tags(),
 	})
 
+	// If internet-facing is restricted to certain namespaces, then validate this is one of them
+	if o.RestrictScheme && *newIngress.annotations.Scheme == "internet-facing" {
```

Comment:
I think we should consider the following with validating whether `alb.ingress.kubernetes.io/scheme=internet-facing` should be rejected.

- Move this logic to the `annotations` package.
  - Inside the following logic https://github.com/coreos/alb-ingress-controller/blob/master/pkg/annotations/annotations.go#L287-L296
  - See how we've done similar for validating / rejecting cert ARNs (https://github.com/coreos/alb-ingress-controller/blob/master/pkg/annotations/annotations.go#L126-L137)
  - The goal here being to contain all annotations parse / validation in one place. Additionally to catch issues with this annotation early.